app/api/quick-buy/[productId]/route.ts

```diff
@@ -0,0 +1,62 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { getRequestUrl } from '@app/utils/server-utils';
+import { getWixClient } from '@app/hooks/useWixClientServer';
+import { cart as cartTypes } from '@wix/ecom';
+
+export async function GET(
+  request: NextRequest,
+  {
+    params: { productId },
+  }: {
+    params: { productId: string };
+  }
+) {
+  const requestUrl = getRequestUrl(request);
+  const baseUrl = new URL('/', requestUrl).toString();
+  const { searchParams } = new URL(requestUrl);
+  const quantity = parseInt(searchParams.get('quantity') || '1', 10);
+  const productOptions = JSON.parse(
+    searchParams.get('productOptions') || 'null'
+  );
+  const wixClient = await getWixClient();
+  const { product } = await wixClient.products.getProduct(productId);
+  if (!product) {
+    return new Response('Product not found', {
+      status: 404,
+    });
+  }
+  const selectedOptions =
+    productOptions ??
+    (product.manageVariants
+      ? { variantId: product.variants![0]._id }
+      : product?.productOptions?.length
+      ? {
+          options:
+            product?.productOptions?.reduce((acc, option) => {
+              acc[option.name!] = option.choices![0].description!;
+              return acc;
+            }, {} as Record<string, any>) ?? {},
+        }
+      : undefined);
+  const item = {
+    quantity,
+    catalogReference: {
+      catalogItemId: product._id!,
+      appId: '1380b703-ce81-ff05-f115-39571d94dfcd',
+      options: selectedOptions,
+    },
+  };
+  const cart = await wixClient.cart.createCart({ lineItems: [item] });
```

Comment:
no need to create a cart in this case. you can create checkout with line items
https://dev.wix.com/api/sdk/ecom/checkout/createcheckout